lib/voxpupuli/acceptance/spec_helper_acceptance.rb

```diff
@@ -1,4 +1,31 @@
-def configure_beaker(modules: :metadata, &block)
+ENV_VAR_PREFIX = 'BEAKER_FACTER_'
+FACT_FILE = '/etc/facter/facts.d/voxpupuli-acceptance-env.json'
+
+def beaker_facts_from_env
+  facts = {}
+
+  ENV.each do |var, value|
+    next unless var.start_with?(ENV_VAR_PREFIX)
+
+    fact = var.sub(ENV_VAR_PREFIX, '').downcase
+    facts[fact] = value
+  end
+
+  facts
+end
+
+def write_beaker_facts_on(hosts)
+  beaker_facts = beaker_facts_from_env
+
+  if beaker_facts.any?
+    require 'json'
+    on(hosts, "mkdir -p #{File.dirname(fact_file)} && cat <<VOXPUPULI_BEAKER_ENV_VARS > #{fact_file}\n#{beaker_facts.to_json}VOXPUPULI_BEAKER_ENV_VARS")
```

Comment:
Since there are no structured facts, you might just loop through them and write each to one a .txt file as opposed to converting to json.

Reply:
One problem is that you can't clean up then. Perhaps a single dedicated variable? Another is to use file prefixes and clean those up, but that requires more coding.

Reply:
What do you mean cleanup? You could put them all in one file like `beaker.txt`.

Reply:
Sorry, that's what I meant with `single dedicated variable`. Should have said `file`.